MyString.java

```diff
@@ -36,9 +41,30 @@ public static int countChar(String str, char ch) {
      * @return true is str1 is a subset of str2, false otherwise
      */
     public static boolean subsetOf(String str1, String str2) {
-         //// Replace the following statement with your code
-        return false;
-    }
+        if ( str2.length() < str1.length()){
+            return false;
+        }
+        int j = 0;
+        boolean isSub = false;
+         for( int i=0; i < str1.length(); i++){
+            j = 0;
+            isSub = false;
+            while(j < str2.length() && !isSub){
+             if(str1.charAt(i) == str2.charAt(j)){
+                isSub = true;
+             }
+               j++;
+                
+            }
+            if(!isSub){
+                return false;
+            }
+            str2 = str2.substring(0, j-1) + str2.substring(j);
+            }
+            return true;
+         }

```

Comment:
Subset doesn't need to check order.. 
You could put some of it in a helper function